docs/configuring-playbook-bridge-mautrix-slack.md

```diff
@@ -0,0 +1,78 @@
+# Setting up Mautrix Slack (optional)
+
+**Note**: bridging to [Slack](https://slack.com/) can also happen via the [mx-puppet-slack](configuring-playbook-bridge-mx-puppet-slack.md) and [matrix-appservice-slack](configuring-playbook-bridge-appservice-slack.md) bridges supported by the playbook.
+- For using as a Bot we recommend the [Appservice Slack](configuring-playbook-bridge-appservice-slack.md), because it supports plumbing.
+- For personal use with a slack account we recommend the `mautrix-slack` bridge (the one being discussed here), because it is the most fully-featured and stable of the 3 Slack bridges supported by the playbook.
+The `mautrix-slack` bridge (the one being discussed here) is the most fully-featured and stable of the 3 Slack bridges supported by the playbook, so it's the one we recommend.
+
+The playbook can install and configure [mautrix-slack](https://github.com/mautrix/slack) for you.
+
+See the project's [documentation](https://docs.mau.fi/bridges/go/slack/index.html) to learn what it does and why it might be useful to you.
+
+Note that as of Oct 2022, support for multiple Matrix users using the bot is incomplete. Different users do not yet share the bridged channels. Everyone gets their own copy.
+
+See the [features and roadmap](https://github.com/mautrix/slack/blob/main/ROADMAP.md) for more information.
+
+
+## Prerequisites
+
+For using this bridge, you would need to authenticate by **providing your username and password** (legacy) or by using a **token login**. See more information in the [docs](https://docs.mau.fi/bridges/go/slack/authentication.html).
+
+Note that neither of these methods are officially supported by Slack. [matrix-appservice-slack](configuring-playbook-bridge-appservice-slack.md) uses a Slack bot account which is the only Slack officially supported method for bridging a channel.
+
+
+## Installing
+
+To enable the bridge, add this to your `vars.yml` file:
+
+```yaml
+matrix_mautrix_slack_enabled: true
+```
+
+You may optionally wish to add some [Additional configuration](#additional-configuration), or to [prepare for double-puppeting](#set-up-double-puppeting) before the initial installation.
+
+After adjusting your `vars.yml` file, re-run the playbook and restart all services: `ansible-playbook -i inventory/hosts setup.yml --tags=setup-all,start`
+
+To make use of the bridge, see [Usage](#usage) below.
+
+
+### Additional configuration
+
+There are some additional things you may wish to configure about the bridge.
+
+Take a look at:
+
+- `roles/matrix-bridge-mautrix-slack/defaults/main.yml` for some variables that you can customize via your `vars.yml` file
+- `roles/matrix-bridge-mautrix-slack/templates/config.yaml.j2` for the bridge's default configuration. You can override settings (even those that don't have dedicated playbook variables) using the `matrix_mautrix_slack_configuration_extension_yaml` variable
```

Comment:
Maybe it's somewhere (and on `docs/configuring-playbook-bridge-appservice-slack.md`) here that we should mention that the `appservice-slack` bridge also uses the same bot username (`slackbot`) by default..

And that if one intends to enable both bridges at the same time, they probably need to change either `matrix_mautrix_slack_appservice_bot_username` (this role) or `matrix_appservice_slack_bot_name` (the `appservice-slack` role).

---

Instead of a notice (which adds to the wall-of-text and may be easily missed), a `fail` task in each of these roles may be **better**. It's a little bit against our rule of not referencing variables from other roles, but ..

**Untested** example (`roles/matrix-bridge-mautrix-slack/tasks/validate_config.yml`):

```yaml
- name: Fail if appservice-slack and mautrix-slack have conflicting bot usernames
  when: matrix_appservice_slack_enabled | default(False) | bool and matrix_mautrix_slack_appservice_bot_username == matrix_appservice_slack_bot_name | default ('')
  ansible.builtin.fail:
    msg: |
      The appservice-slack and mautrix-slack components are both enabled and use the same bot username ({{ matrix_mautrix_slack_appservice_bot_username }}), as per their default configuration, which causes a conflcit.
      To resolve the conflict, make one of these components use a different username.
      Consider either changing `matrix_mautrix_slack_appservice_bot_username` (the bot username for the mautrix-slack component) or `matrix_appservice_slack_bot_name` (the bot username for the appservice-slack component).
      We recommend that you change the username for the newly-added (and yet unused) component.
```

In `roles/matrix-bridge-appservice-slack/tasks/validate_config.yml` we'd need the same thing but with a different `when`:

```yaml
when: matrix_mautrix_slack_enabled | default(False) | bool and matrix_mautrix_slack_appservice_bot_username | default ('') == matrix_appservice_slack_bot_name
```

--

^^ We use ` | default(..)` above to allow the roles to not have a hard dependency on one another.

Reply:
Alternatively, we could make the new role (`mautrix-slack`) use s a different username by default.

Reply:
Good catch, added the checks. Do we want to additionally change the default username?

Reply:
Sorry for somehow missing your reply here. It's just now that I see it.

While I'd love for the playbook to be as conflict-free as possible by default (that is, you can enable and disable stuff as you wish at any time and not worry about conflicts), I'd also like to stick to upstream defaults as much as possible. Otherwise, we might get people reading the the [upstream mautrix-slack Authentication page](https://docs.mau.fi/bridges/go/slack/authentication.html) and trying to message `@slackbot`. If we've renamed it to `@mautrixslackbot`, we might get contacted by people saying "it's not working".

I'm not sure how many people will follow these upstream docs.. And how many will also disregard our own docs and warnings. Probably only a few, if at all?

Maybe deviating from upstream defaults is better in this case. If upstream wanted consistency, they should have avoided reusing names from other (older) projects.

If we go for this new default username, we could probably get rid of these conflict checks in both `validate_config.ym` files.

Reply:
I think in-line with upstream sounds like the better priority than user config as it is now then. Keeping the checks in and username as default.